dht/sharder.hh

```diff
@@ -28,28 +28,67 @@
 
 namespace dht {
 
+
+// Utilities for sharding ring partition_range:s
+
+// A ring_position range's data is divided into sub-ranges, where each sub-range's data
+// is owned by a single shard. Note that multiple non-overlapping sub-ranges may map to a
+// single range, and some shards may not receive any sub-range.
+//
+// This module provides utilities for determining the sub-ranges to shard mapping. The utilities
+// generate optimal mappings: each range that you get is the largest possible, so you
+// get the minimum number of ranges possible. You can get many ranges, so operate on them
+// one (or a few) at a time, rather than accumulating them.
+
+// A mapping between a partition_range and a shard. All positions within `ring_range` are
+// owned by `shard`.
+//
+// The classes that return ring_position_range_and_shard make them as large as possible,
```

Comment:
It would be more clear to say 'make ring_range as large as possible' instead of 'make them as large as possible'.

Reply:
I believe that it would be valuable to explain what it means for a range to be as large as possible. It means that enlarging it at any end will mean the range is no longer owned by a single shard. I know this can be deduced but let's not put too much pressure on the reader ;)

Reply:
Done.